src/java.base/share/classes/java/lang/invoke/MethodType.java

```diff
@@ -1220,19 +1222,19 @@ static MethodType fromDescriptor(String descriptor, ClassLoader loader)
     }
 
     /**
-     * Returns a descriptor string for the method type.  This method
+     * {@return the descriptor string for this method type} This method
      * is equivalent to calling {@link #descriptorString() MethodType::descriptorString}.
      *
-     * <p>
-     * Note that this is not a strict inverse of {@link #fromMethodDescriptorString fromMethodDescriptorString}.
-     * Two distinct classes which share a common name but have different class loaders
-     * will appear identical when viewed within descriptor strings.
```

Comment:
Re-reading this, I think this api note reads fine.   OTOH I find a little confusing how you rephrased it.  "A method type produced by changing a component class object to...".   I understand you tried to make it crystal clear two method types may have the identical descriptor strings even if its parameter type or return type are distinct Class objects.   

I think the original version is clear enough and less lengthy.

Reply:
```
     * Two distinct classes which share a common name but have different class loaders
     * produce identical component descriptors in descriptor strings.
```
Does this sound good? This also addresses alan's wording concerns.

Reply:
Why can't just say "identical descriptor strings"?   It's unclear what "component descriptor" is.

Reply:
"component descriptor" tried to refer to both parameter types and return type descriptors. How about "produce identical class descriptors in method descriptors?"

Reply:
`MethodType::descriptorString` refers to method descriptor and `Class::descriptorString` refers to class descriptor.   I don't think it helps to say "produce identical class descriptors in method descriptors".

Reply:
I think we can put it simple:
"Two distinct classes which share a common name but have different class loaders cannot be distinguished in descriptor strings"
which stands for both class and method descriptors. How does this sound?

Reply:
What about "Two distinct {@code MethodType} objects can have an identical descriptor string as distinct classes can have the same name but different class loaders."?

This api note basically says that from a given descriptor string, `fromMethodDescriptorString` may produce a different `MethodType` with a different class loader.